README.md
# Quotes Widget

QuotesWidget is a package used to display a random quote enclosed by a neumorphic container.

![Image](https://github.com/sivaprasadnk/quotes_widget/blob/main/screenshots/Screenshot.png)
## Getting Started

Add this to your package's `pubspec.yaml` file

```yaml
dependencies:
  quotes_widget: ^0.0.2
```

## Usage

Next, you just have to import the package using:

```dart
import 'package:quotes_widget/quotes_widget.dart';
```


```dart
  Widget build(BuildContext context) {
    return QuotesWidget(
        width: 300,
        height: 200,
        quoteFontSize: 21,
        authorFontSize: 18,
    ),
  }
```
## Contributing
Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.


CHANGELOG.md
## 0.0.2

* Correcting Documentation.



test/quotes_widget_test.dart
// import 'package:flutter_test/flutter_test.dart';

// import 'package:quotes_widget/quotes_widget.dart';

// void main() {
//   test('adds one to input values', () {
//     final calculator = Calculator();
//     expect(calculator.addOne(2), 3);
//     expect(calculator.addOne(-7), -6);
//     expect(calculator.addOne(0), 1);
//   });
// }


lib/quotes_widget.dart
import 'dart:math';

import 'package:flutter/material.dart';
import 'package:glass_kit/glass_kit.dart';
import 'package:quotes_widget/src/quotes_list.dart';

/// A simple widget to showcase quote and their author surrounded by a neumorphic container
///
/// Generates a random quote from predefined list and shows in UI.
///
/// [height] and [width] values must not be null.
///
class QuotesWidget extends StatelessWidget {
  const QuotesWidget({
    Key? key,
    required this.height,
    required this.width,
    required this.quoteFontSize,
    required this.authorFontSize,
  }) : super(key: key);

  /// Height of the neumorphic container
  final double height;

  /// Width of the neumorphic container
  final double width;

  /// font-size of quote text
  final double quoteFontSize;

  ///font-size of author-name
  final double authorFontSize;

  @override
  Widget build(BuildContext context) {
    int length = quotesList.length;

    /// to generate random number
    var r = Random().nextInt(length - 1);
    return Stack(
      children: [
        Padding(
          padding: const EdgeInsets.only(top: 12),
          child: GlassContainer(
            height: height,
            width: width,
            borderRadius: BorderRadius.circular(15),
            gradient: LinearGradient(
              colors: [
                Colors.white.withOpacity(0.40),
                Colors.white.withOpacity(0.10)
              ],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
            ),
            borderColor: Colors.cyanAccent,
            borderGradient: LinearGradient(
              colors: [
                Colors.white.withOpacity(0.60),
                Colors.white.withOpacity(0.10),
                Colors.lightBlueAccent.withOpacity(0.05),
                Colors.lightBlueAccent.withOpacity(0.6)
              ],
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              stops: const [0.0, 0.39, 0.40, 1.0],
            ),
            blur: 10.0,
            borderWidth: 3,
            elevation: 5.0,
            isFrostedGlass: false,
            shadowColor: Colors.black.withOpacity(0.20),
            alignment: Alignment.center,
            frostedOpacity: 0.12,
            margin: const EdgeInsets.all(8.0),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              mainAxisSize: MainAxisSize.min,
              children: [
                const Spacer(),
                Padding(
                  padding: const EdgeInsets.only(left: 20, right: 20, top: 10),
                  child: Text(
                    quotesList[r],
                    style: TextStyle(
                      fontSize: quoteFontSize,
                      color: Colors.white,
                      fontFamily: 'Vollkorn',
                      package: 'quotes_widget',
                    ),
                  ),
                ),
                Row(
                  mainAxisAlignment: MainAxisAlignment.end,
                  children: [
                    Text(
                      '\n- ${authorsList[r]}',
                      textAlign: TextAlign.end,
                      style: TextStyle(
                        fontSize: authorFontSize,
                        color: Colors.white,
                      ),
                    ),
                    const SizedBox(
                      width: 20,
                    )
                  ],
                ),
                const Spacer(),
                // const Spacer(),
              ],
            ),
          ),
        ),
        const Positioned.fill(
          left: 50,
          child: Align(
            alignment: Alignment.topLeft,
            child: Image(
              height: 50,
              image: AssetImage(
                "assets/images/quote.png",
                package: "quotes_widget",
              ),
            ),
          ),
        )
      ],
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:quotes_widget/quotes_widget.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const Home());
  }
}

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const Scaffold(
      backgroundColor: Colors.blueAccent,
      body: Center(
        child: QuotesWidget(
          width: 300,
          height: 200,
          quoteFontSize: 21,
          authorFontSize: 18,
        ),
      ),
    );
  }
}
